### DISCUSSION TITLE
Oauth2 via ".authWithOAuth2" with Flutter and iOS

### DISCUSSION THREAD

**Discussion by agreensh** [pocketbase/pocketbase#1890] [Q&A]
I am calling 

    final authMethods = await client.collection('users').listAuthMethods();
    final google = authMethods.authProviders.where((am) => am.name.toLowerCase() == "google").first;

    final result = await FlutterWebAuth2.authenticate(url: url.toString(), callbackUrlScheme: callbackUrlScheme);

to fetch the 'code' parameter required below. This call is successful.

However, when I call "client.collection('users').authWithOAuth2('google', code, google.codeVerifier, 'pocketbase.scheme');" (pocketbase.scheme is my redirect scheme defined in the plist file), I just get a 400 with no useful error message - "Something went wrong while processing your request."
Anyone have suggestions or successfully used authWithOAuth2 on iOS?

**Comment by ganigeorgiev** [comment]
You can start the PocketBase executable with `--debug` flag and check for a more detailed error message in the console.
Alternatively, you can also inspect the error request log in the "Admin UI > Logs".

**Reply by ganigeorgiev** [reply]
Also note that the sign-in/sign-up url must be concatenated with the redirect url, aka. `url` in the above must be something like: `google.authUrl + "pocketbase.scheme"`.

As a reference, if you haven't already, you can also check the example web integration https://pocketbase.io/docs/authentication/#web-oauth2-integration (_the expected flow should be more or less similar with an exeception maybe the redirect/deeplink handling_).

**Comment by agreensh** [comment]
Thanks for the tips.
The error is: 

oauth2: cannot fetch token: 401 Unauthorized
Response: {
  "error": "invalid_client",
  "error_description": "Unauthorized"
}

I don't understand what this means - I have checked the client_id etc, and everything is correct.
If I call Google directly, with the 'code' I get from the above call, to get a token, that works fine

    final url = Uri.https('www.googleapis.com', 'oauth2/v4/token');
    final response = await http.post(url, body: {
      'client_id': clientId,
      'redirect_uri': redirectUri,
      'grant_type': 'authorization_code',
      'code': code,
    });

Do you have anymore thoughts/suggestions?

 Note: a Google search suggests this error can occur if the header "content-type" is used, rather than the expected "Content-type"

**Reply by ganigeorgiev** [reply]
I'm not able to reproduce the issue locally using the previously linked [plain JS example](https://pocketbase.io/docs/authentication/#web-oauth2-integration). Make sure that your Google client id and secret are correct.

What is `url.toString()` in the initial example? Are you constructing manually the sign-in url or using the one from the `listAuthMethods` call? Could you provide a more detailed code sample (or better a minimal reproducible repo)?
